src/main/kotlin/step2/calculator/Calculator.kt

```diff
@@ -0,0 +1,34 @@
+package step2.calculator
+
+import step2.calculator.StringExpression.parseToMathExpression
+
+object Calculator {
+    private const val ARITHMETIC_EXCEPTION_ERROR_MESSAGE = "Error occurred during calculate"
+
+    fun calculate(expression: String): Int {
+        val mathExpression = parseToMathExpression(expression)
+
+        return kotlin.runCatching {
+            calculate(mathExpression)
+        }.onFailure {
+            throw ArithmeticException(ARITHMETIC_EXCEPTION_ERROR_MESSAGE)
+        }.getOrThrow()
```

Comment:
```suggestion
        }.getOrElse {
            throw ArithmeticException(ARITHMETIC_EXCEPTION_ERROR_MESSAGE)
        }
```
getOrElse를 활용할 수 있을 것 같아요

Reply:
onFailure는 Result객체를 받는 쪽에서 사용할때 주로 사용합니다.
현재 Result 타입으로 wrapping하는 것이 아니기 때문에, getOrElse가 더 적합한 것 같아요

Reply:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/get-or-throw.html

위 문서에 보면 `This function is a shorthand for getOrElse { throw it } (see getOrElse).`
약어라고 설명이 나오는데요
이해가 잘 안되서 그런데 다시 설명을 부탁드려도 될까요?

Reply:
일단 `getOrElse`로 변경하였습니다~